frontend/src/routes/Search.tsx

```diff
@@ -15,41 +22,65 @@ import { SmallDescription } from "../ui/metadata";
 import { Card } from "../ui/Card";
 import { Breadcrumbs, BreadcrumbsContainer, BreadcrumbSeparator } from "../ui/Breadcrumbs";
 import { MissingRealmName } from "./util";
-import { ellipsisOverflowCss } from "../ui";
+import { ellipsisOverflowCss, focusStyle } from "../ui";
 import { COLORS } from "../color";
 import { BREAKPOINT_MEDIUM, BREAKPOINT_SMALL } from "../GlobalStyle";
-import { keyOfId } from "../util";
+import { isExperimentalFlagSet, keyOfId } from "../util";
+import { Button } from "../ui/Button";
 import { DirectVideoRoute, VideoRoute } from "./Video";
 
 
 export const isSearchActive = (): boolean => document.location.pathname === "/~search";
 
+type SearchParams = {
+    query: string;
+    filter: string | undefined;
+    start: string | undefined;
+    end: string | undefined;
+}
+
 export const SearchRoute = makeRoute({
-    url: ({ query }: { query: string }) => `/~search?${new URLSearchParams({ q: query })}`,
+    url: ({ query, filter, start, end }: SearchParams) => {
+        const searchParams = new URLSearchParams({ q: query });
+
+        if (filter) {
+            searchParams.append("f", filter);
+        }
+        if (start) {
+            searchParams.append("start", start);
+        }
+        if (end) {
+            searchParams.append("end", end);
+        }
+
+        return `/~search?${searchParams}`;
+    },
     match: url => {
         if (url.pathname !== "/~search") {
             return null;
         }
 
-        const q = url.searchParams.get("q") ?? "";
-        const queryRef = loadQuery<SearchQuery>(query, { q });
+        const [q, f, start, end] = ["q", "f", "start", "end"].map(param =>
+            url.searchParams.get(param) ?? "");
```

Comment:
The default value shouldn't be empty string for those parameters. Rather, they should be `null` or `undefined`. Needs adjustments in the API I think. But I mentioned that before: that all parameters (except q) should be optional.